arbor/include/arbor/cable_cell_param.hpp

```diff
@@ -46,55 +46,87 @@ struct i_clamp {
     i_clamp(value_type delay, value_type duration, value_type amplitude):
         delay(delay), duration(duration), amplitude(amplitude)
     {}
+
+    friend std::ostream& operator<<(std::ostream& o, const i_clamp& p) {
+        return o << "(current-clamp " << p.delay << ' ' << p.duration << ' ' << p.amplitude << ')';
```

Comment:
Seeing these makes me wish for `fmt` or `std::format`. Maybe these can be made less verbose by using `util::pprintf`?

On the other hand, did you consider to have a method `s_expr to_expr(const i_clamp&)` (etc) and delegating the actual printing part to the `s_expr` functionality? I believe @halfflat mentioned something similar.

Reply:
These methods aren't really needed in the implementation, so they can be removed. I kept them in because I thought they could be useful. I can use util::pprintf or just remove them completely. 
We do have `mksexp` functions in the `arborio` lib, but I didn't want s-expressions to bleed too much into the core arbor library (especially public headers) because I'd rather we move the `s_expr` implementation completely into `arborio` eventually. 
@halfflat's suggetsion would be useful here, but I think it'll take some more work, which I will do in another PR after we discuss the details a bit more.

Reply:
I think it's worth keeping them. Our python bindings could make use of them to implement `__str__`/`__repr__`.

Reply:
Agreed on the rest.